crates/turbopack-ecmascript/src/parse/top_level_await.rs

```diff
@@ -0,0 +1,17 @@
+use swc_core::ecma::{ast::*, visit::Visit};
+
+#[derive(Default)]
+pub(super) struct TopLevelAwaitVisitor {
+    pub(super) has_top_level_await: bool,
+}
+
+impl Visit for TopLevelAwaitVisitor {
+    fn visit_await_expr(&mut self, _: &AwaitExpr) {
+        self.has_top_level_await = true;
+    }
+
+    // prevent non top level items from visiting their children
+    fn visit_arrow_expr(&mut self, _: &ArrowExpr) {}
+    fn visit_class(&mut self, _: &Class) {}
```

Comment:
Class can have it in a few positions, like the `extends (await foo)`, `static x = await foo`, `[await foo] = 1;`. You need to skip the class instance property value, and method bodies.